mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseToStdlib.cpp

```diff
@@ -246,18 +246,51 @@ struct BufferLoadRewritePattern : public OpRewritePattern<BufferLoadOp> {
         oobDims.push_back(pair.index());
     bool toEmitOobChecks = !oobDims.empty();
 
-    auto emitLoadInstruction = [&b, loc, loadedType,
+    auto emitLoadInstruction = [&b, loc, loadedType, sourceType,
                                 source](ValueRange loadCoords) -> Value {
       Value loadedValue;
-      if (auto loadedVectorType = loadedType.dyn_cast<VectorType>()) {
-        // Issue vector load.
-        // use buffer load since the source memref is on address space 0
-        SmallVector<Value, 4> srcLowerIndicesI32;
-        for (auto v : loadCoords)
-          srcLowerIndicesI32.push_back(
-              b.create<IndexCastOp>(loc, b.getIntegerType(32), v));
-        loadedValue = b.create<gpu::MubufLoadOp>(loc, loadedType, source,
-                                                 srcLowerIndicesI32);
+
+      Type elementType = loadedType;
+      int64_t vectorLength = 1;
+
+      if (loadedType.isa<VectorType>()) {
+        VectorType loadedVectorType = loadedType.cast<VectorType>();
+        elementType = loadedVectorType.getElementType();
+        vectorLength = loadedVectorType.getShape()[0];
+
+        auto loadWidth = vectorLength * elementType.getIntOrFloatBitWidth();
+        bool isTooNarrow = loadWidth < 32;
+        bool isTooWide = loadWidth > 4 * 32;
+
+        // Use scalar load when load width is too narrow or too wide
+        // for a mubuf instruction
+        if (isTooNarrow || isTooWide) {
```

Comment:
Is there any test case in the existing test suite to cover this code path?

Reply:
@whchung Yes, with the tuning parameters in this PR, if this chunk of code doesn't exist, it would directly yield into a lowering failure that Mubuf instruction on 2 i8 is wrong for a number of resnet50 configs. (And now all resnet50 configs are passing)

Below config is how I used to test and bring up:

> ./bin/miopen-gen -batchsize=64 -in_channels=128 -in_h=16 -in_w=16 -out_channels=128 -fil_h=1 -fil_w=1 --dilation_h=1 --dilation_w=1 --conv_stride_h=1 --conv_stride_w=1 --padding_h=1 --padding_w=1 -t i8 -x2 --operation conv2d -pv | ./bin/mlir-miopen-driver -c | ./bin/mlir-rocm-runner --shared-libs=./lib/librocm-runtime-wrappers.so,./external/llvm-project/llvm/lib/libmlir_runner_utils.so --entry-point-result=void